File: qt/network/gen_qnetworkrequestfactory.go
package network

import (
	"github.com/mappu/miqt/qt"
	"unsafe"
)

type QNetworkRequestFactory struct {
	h          uintptr
	isSubclass bool
}

// NewQNetworkRequestFactory constructs a new QNetworkRequestFactory object.
func NewQNetworkRequestFactory() *QNetworkRequestFactory {

	ret := newQNetworkRequestFactory(QNetworkRequestFactory_new())
	ret.isSubclass = true
	return ret
}

// NewQNetworkRequestFactory2 constructs a new QNetworkRequestFactory object.
func NewQNetworkRequestFactory2(baseUrl *qt.QUrl) *QNetworkRequestFactory {

	ret := newQNetworkRequestFactory(QNetworkRequestFactory_new2((*QUrl)(baseUrl.UnsafePointer())))
	ret.isSubclass = true
	return ret
}

// NewQNetworkRequestFactory3 constructs a new QNetworkRequestFactory object.
func NewQNetworkRequestFactory3(other *QNetworkRequestFactory) *QNetworkRequestFactory {

	ret := newQNetworkRequestFactory(QNetworkRequestFactory_new3(other.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QNetworkRequestFactory) OperatorAssign(other *QNetworkRequestFactory) {
	QNetworkRequestFactory_OperatorAssign(this.h, other.cPointer())
}

func (this *QNetworkRequestFactory) Swap(other *QNetworkRequestFactory) {
	QNetworkRequestFactory_Swap(this.h, other.cPointer())
}

func (this *QNetworkRequestFactory) BaseUrl() *qt.QUrl {
	_goptr := qt.UnsafeNewQUrl(unsafe.Pointer(QNetworkRequestFactory_BaseUrl(this.h)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) SetBaseUrl(url *qt.QUrl) {
	QNetworkRequestFactory_SetBaseUrl(this.h, (*QUrl)(url.UnsafePointer()))
}

func (this *QNetworkRequestFactory) SslConfiguration() *QSslConfiguration {
	_goptr := newQSslConfiguration(QNetworkRequestFactory_SslConfiguration(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) SetSslConfiguration(configuration *QSslConfiguration) {
	QNetworkRequestFactory_SetSslConfiguration(this.h, configuration.cPointer())
}

func (this *QNetworkRequestFactory) CreateRequest() *QNetworkRequest {
	_goptr := newQNetworkRequest(QNetworkRequestFactory_CreateRequest(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) CreateRequestWithQuery(query *qt.QUrlQuery) *QNetworkRequest {
	_goptr := newQNetworkRequest(QNetworkRequestFactory_CreateRequestWithQuery(this.h, (*QUrlQuery)(query.UnsafePointer())))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) CreateRequestWithPath(path string) *QNetworkRequest {
	path_ms := struct_miqt_string{}
	path_ms.data = CString(path)
	path_ms.len = size_t(len(path))
	defer free(unsafe.Pointer(path_ms.data))
	_goptr := newQNetworkRequest(QNetworkRequestFactory_CreateRequestWithPath(this.h, path_ms))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) CreateRequest2(path string, query *qt.QUrlQuery) *QNetworkRequest {
	path_ms := struct_miqt_string{}
	path_ms.data = CString(path)
	path_ms.len = size_t(len(path))
	defer free(unsafe.Pointer(path_ms.data))
	_goptr := newQNetworkRequest(QNetworkRequestFactory_CreateRequest2(this.h, path_ms, (*QUrlQuery)(query.UnsafePointer())))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) SetCommonHeaders(headers *QHttpHeaders) {
	QNetworkRequestFactory_SetCommonHeaders(this.h, headers.cPointer())
}

func (this *QNetworkRequestFactory) CommonHeaders() *QHttpHeaders {
	_goptr := newQHttpHeaders(QNetworkRequestFactory_CommonHeaders(this.h))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) ClearCommonHeaders() {
	QNetworkRequestFactory_ClearCommonHeaders(this.h)
}

func (this *QNetworkRequestFactory) BearerToken() []byte {
	var _bytearray struct_miqt_string = QNetworkRequestFactory_BearerToken(this.h)
	_ret := GoBytes(unsafe.Pointer(_bytearray.data), int(int64(_bytearray.len)))
	free(unsafe.Pointer(_bytearray.data))
	return _ret
}

func (this *QNetworkRequestFactory) SetBearerToken(token []byte) {
	token_alias := struct_miqt_string{}
	token_alias.data = (char)(unsafe.Pointer(&token[0]))
	token_alias.len = size_t(len(token))
	QNetworkRequestFactory_SetBearerToken(this.h, token_alias)
}

func (this *QNetworkRequestFactory) ClearBearerToken() {
	QNetworkRequestFactory_ClearBearerToken(this.h)
}

func (this *QNetworkRequestFactory) UserName() string {
	var _ms struct_miqt_string = QNetworkRequestFactory_UserName(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QNetworkRequestFactory) SetUserName(userName string) {
	userName_ms := struct_miqt_string{}
	userName_ms.data = CString(userName)
	userName_ms.len = size_t(len(userName))
	defer free(unsafe.Pointer(userName_ms.data))
	QNetworkRequestFactory_SetUserName(this.h, userName_ms)
}

func (this *QNetworkRequestFactory) ClearUserName() {
	QNetworkRequestFactory_ClearUserName(this.h)
}

func (this *QNetworkRequestFactory) Password() string {
	var _ms struct_miqt_string = QNetworkRequestFactory_Password(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QNetworkRequestFactory) SetPassword(password string) {
	password_ms := struct_miqt_string{}
	password_ms.data = CString(password)
	password_ms.len = size_t(len(password))
	defer free(unsafe.Pointer(password_ms.data))
	QNetworkRequestFactory_SetPassword(this.h, password_ms)
}

func (this *QNetworkRequestFactory) ClearPassword() {
	QNetworkRequestFactory_ClearPassword(this.h)
}

func (this *QNetworkRequestFactory) QueryParameters() *qt.QUrlQuery {
	_goptr := qt.UnsafeNewQUrlQuery(unsafe.Pointer(QNetworkRequestFactory_QueryParameters(this.h)))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) SetQueryParameters(query *qt.QUrlQuery) {
	QNetworkRequestFactory_SetQueryParameters(this.h, (*QUrlQuery)(query.UnsafePointer()))
}

func (this *QNetworkRequestFactory) ClearQueryParameters() {
	QNetworkRequestFactory_ClearQueryParameters(this.h)
}

func (this *QNetworkRequestFactory) SetPriority(priority QNetworkRequest__Priority) {
	QNetworkRequestFactory_SetPriority(this.h, (int)(priority))
}

func (this *QNetworkRequestFactory) Priority() QNetworkRequest__Priority {
	return (QNetworkRequest__Priority)(QNetworkRequestFactory_Priority(this.h))
}

func (this *QNetworkRequestFactory) Attribute(attribute QNetworkRequest__Attribute) *qt.QVariant {
	_goptr := qt.UnsafeNewQVariant(unsafe.Pointer(QNetworkRequestFactory_Attribute(this.h, (int)(attribute))))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) Attribute2(attribute QNetworkRequest__Attribute, defaultValue *qt.QVariant) *qt.QVariant {
	_goptr := qt.UnsafeNewQVariant(unsafe.Pointer(QNetworkRequestFactory_Attribute2(this.h, (int)(attribute), (*QVariant)(defaultValue.UnsafePointer()))))
	_goptr.GoGC() // Qt uses pass-by-value semantics for this type. Mimic with finalizer
	return _goptr
}

func (this *QNetworkRequestFactory) SetAttribute(attribute QNetworkRequest__Attribute, value *qt.QVariant) {
	QNetworkRequestFactory_SetAttribute(this.h, (int)(attribute), (*QVariant)(value.UnsafePointer()))
}

func (this *QNetworkRequestFactory) ClearAttribute(attribute QNetworkRequest__Attribute) {
	QNetworkRequestFactory_ClearAttribute(this.h, (int)(attribute))
}

func (this *QNetworkRequestFactory) ClearAttributes() {
	QNetworkRequestFactory_ClearAttributes(this.h)
}

File: qt/network/gen_qnetworkrequestfactory.cpp
// +build ignore

#include <QByteArray>
#include <QHttpHeaders>
#include <QNetworkRequest>
#include <QNetworkRequestFactory>
#include <QSslConfiguration>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <QUrlQuery>
#include <QVariant>
#include <qnetworkrequestfactory.h>
#include "gen_qnetworkrequestfactory.h"
QNetworkRequestFactory* QNetworkRequestFactory_new() {
return new QNetworkRequestFactory();
}
QNetworkRequestFactory* QNetworkRequestFactory_new2(QUrl* baseUrl) {
return new QNetworkRequestFactory(*baseUrl);
}
QNetworkRequestFactory* QNetworkRequestFactory_new3(QNetworkRequestFactory* other) {
return new QNetworkRequestFactory(*other);
}
void QNetworkRequestFactory_OperatorAssign(QNetworkRequestFactory* self, QNetworkRequestFactory* other) {
	self->operator=(*other);
}
void QNetworkRequestFactory_Swap(QNetworkRequestFactory* self, QNetworkRequestFactory* other) {
	self->swap(*other);
}
QUrl* QNetworkRequestFactory_BaseUrl(const QNetworkRequestFactory* self) {
	return new QUrl(self->baseUrl());
}
void QNetworkRequestFactory_SetBaseUrl(QNetworkRequestFactory* self, QUrl* url) {
	self->setBaseUrl(*url);
}
QSslConfiguration* QNetworkRequestFactory_SslConfiguration(const QNetworkRequestFactory* self) {
	return new QSslConfiguration(self->sslConfiguration());
}
void QNetworkRequestFactory_SetSslConfiguration(QNetworkRequestFactory* self, QSslConfiguration* configuration) {
	self->setSslConfiguration(*configuration);
}
QNetworkRequest* QNetworkRequestFactory_CreateRequest(const QNetworkRequestFactory* self) {
	return new QNetworkRequest(self->createRequest());
}
QNetworkRequest* QNetworkRequestFactory_CreateRequestWithQuery(const QNetworkRequestFactory* self, QUrlQuery* query) {
	return new QNetworkRequest(self->createRequest(*query));
}
QNetworkRequest* QNetworkRequestFactory_CreateRequestWithPath(const QNetworkRequestFactory* self, struct miqt_string path) {
	QString path_QString = QString::fromUtf8(path.data, path.len);
	return new QNetworkRequest(self->createRequest(path_QString));
}
QNetworkRequest* QNetworkRequestFactory_CreateRequest2(const QNetworkRequestFactory* self, struct miqt_string path, QUrlQuery* query) {
	QString path_QString = QString::fromUtf8(path.data, path.len);
	return new QNetworkRequest(self->createRequest(path_QString, *query));
}
void QNetworkRequestFactory_SetCommonHeaders(QNetworkRequestFactory* self, QHttpHeaders* headers) {
	self->setCommonHeaders(*headers);
}
QHttpHeaders* QNetworkRequestFactory_CommonHeaders(const QNetworkRequestFactory* self) {
	return new QHttpHeaders(self->commonHeaders());
}
void QNetworkRequestFactory_ClearCommonHeaders(QNetworkRequestFactory* self) {
	self->clearCommonHeaders();
}
struct miqt_string QNetworkRequestFactory_BearerToken(const QNetworkRequestFactory* self) {
	QByteArray _qb = self->bearerToken();
	struct miqt_string _ms;
	_ms.len = _qb.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _qb.data(), _ms.len);
	return _ms;
}
void QNetworkRequestFactory_SetBearerToken(QNetworkRequestFactory* self, struct miqt_string token) {
	QByteArray token_QByteArray(token.data, token.len);
	self->setBearerToken(token_QByteArray);
}
void QNetworkRequestFactory_ClearBearerToken(QNetworkRequestFactory* self) {
	self->clearBearerToken();
}
struct miqt_string QNetworkRequestFactory_UserName(const QNetworkRequestFactory* self) {
	QString _ret = self->userName();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QNetworkRequestFactory_SetUserName(QNetworkRequestFactory* self, struct miqt_string userName) {
	QString userName_QString = QString::fromUtf8(userName.data, userName.len);
	self->setUserName(userName_QString);
}
void QNetworkRequestFactory_ClearUserName(QNetworkRequestFactory* self) {
	self->clearUserName();
}
struct miqt_string QNetworkRequestFactory_Password(const QNetworkRequestFactory* self) {
	QString _ret = self->password();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QNetworkRequestFactory_SetPassword(QNetworkRequestFactory* self, struct miqt_string password) {
	QString password_QString = QString::fromUtf8(password.data, password.len);
	self->setPassword(password_QString);
}
void QNetworkRequestFactory_ClearPassword(QNetworkRequestFactory* self) {
	self->clearPassword();
}
QUrlQuery* QNetworkRequestFactory_QueryParameters(const QNetworkRequestFactory* self) {
	return new QUrlQuery(self->queryParameters());
}
void QNetworkRequestFactory_SetQueryParameters(QNetworkRequestFactory* self, QUrlQuery* query) {
	self->setQueryParameters(*query);
}
void QNetworkRequestFactory_ClearQueryParameters(QNetworkRequestFactory* self) {
	self->clearQueryParameters();
}
void QNetworkRequestFactory_SetPriority(QNetworkRequestFactory* self, int priority) {
	self->setPriority(static_cast<QNetworkRequest::Priority>(priority));
}
int QNetworkRequestFactory_Priority(const QNetworkRequestFactory* self) {
	QNetworkRequest::Priority _ret = self->priority();
	return static_cast<int>(_ret);
}
QVariant* QNetworkRequestFactory_Attribute(const QNetworkRequestFactory* self, int attribute) {
	return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute)));
}
QVariant* QNetworkRequestFactory_Attribute2(const QNetworkRequestFactory* self, int attribute, QVariant* defaultValue) {
	return new QVariant(self->attribute(static_cast<QNetworkRequest::Attribute>(attribute), *defaultValue));
}
void QNetworkRequestFactory_SetAttribute(QNetworkRequestFactory* self, int attribute, QVariant* value) {
	self->setAttribute(static_cast<QNetworkRequest::Attribute>(attribute), *value);
}
void QNetworkRequestFactory_ClearAttribute(QNetworkRequestFactory* self, int attribute) {
	self->clearAttribute(static_cast<QNetworkRequest::Attribute>(attribute));
}
void QNetworkRequestFactory_ClearAttributes(QNetworkRequestFactory* self) {
	self->clearAttributes();
}
void QNetworkRequestFactory_Delete(QNetworkRequestFactory* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QNetworkRequestFactory*>( self );
	} else {
		delete self;
	}
}
